packages/snaps-utils/src/manifest/manifest.ts

```diff
@@ -199,35 +197,38 @@ export async function checkManifest(
 export function fixManifest(
   snapFiles: SnapFiles,
   error: ProgrammaticallyFixableSnapError,
-): SnapManifest {
-  const { manifest, packageJson, sourceCode } = snapFiles;
-  const manifestCopy = deepClone(manifest);
+): VirtualFile<SnapManifest> {
+  const { manifest, packageJson } = snapFiles;
+  const clonedFile = manifest.clone();
+  const manifestCopy = clonedFile.result;
 
   switch (error.reason) {
     case SnapValidationFailureReason.NameMismatch:
-      manifestCopy.source.location.npm.packageName = packageJson.name;
+      manifestCopy.source.location.npm.packageName = packageJson.result.name;
       break;
 
     case SnapValidationFailureReason.VersionMismatch:
-      manifestCopy.version = packageJson.version;
+      manifestCopy.version = packageJson.result.version;
       break;
 
     case SnapValidationFailureReason.RepositoryMismatch:
-      manifestCopy.repository = packageJson.repository
-        ? deepClone(packageJson.repository)
+      manifestCopy.repository = packageJson.result.repository
+        ? deepClone(packageJson.result.repository)
         : undefined;
       break;
 
     case SnapValidationFailureReason.ShasumMismatch:
-      manifestCopy.source.shasum = getSnapSourceShasum(sourceCode);
+      manifestCopy.source.shasum = getSnapChecksum(snapFiles);
       break;
 
     /* istanbul ignore next */
     default:
       assertExhaustive(error.reason);
   }
 
-  return manifestCopy;
+  clonedFile.result = manifestCopy;
+  // TODO: Do we need to update clonedFile.value?
```

Comment:
We should probably look into this before merging?

Reply:
We can easily update it by stringifying the result. I just wasn't sure if it was worth it because it isn't used. 

It is probably safer to just do it though, that way we don't forget about the assumption.